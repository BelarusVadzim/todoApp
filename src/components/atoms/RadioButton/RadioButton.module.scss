@import "src/variables.scss";

.radioButton {
  /* HIDE RADIO */
  [type="radio"] { 
    position: absolute;
    opacity: 0;
    width: 0;
    height: 0;
  }

  /* NORMAL STYLES */
  [type="radio"] + span {
    cursor: pointer;
    color: $secondary-font-color;
    font-size: inherit;
    font-family: $main-font;
  }

  [type="radio"] + span:hover {
    color: $primary-font-color;
  }

  /* CHECKED STYLES */
  [type="radio"]:checked + span {
    color: $primary-color;
  }
}
