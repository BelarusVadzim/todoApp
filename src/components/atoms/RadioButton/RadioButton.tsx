import React, { ChangeEvent } from 'react';
import { PropsWithChildren } from 'react';
import style from './RadioButton.module.scss';
import { PropsWithClassName } from 'types';

export type RadioButtonProps = PropsWithChildren &
  PropsWithClassName & {
    groupName: string;
    selectedValue?: string;
    onChange?: (value: string) => void;
    id?: string;
    value: string;
  };

const RadioButton: React.FC<RadioButtonProps> = ({
  children,
  groupName,
  selectedValue,
  onChange,
  id,
  value,
  className = '',
}) => {
  const combinedClassName = `${style.radioButton} ${className}`;
  const inputOnChange = (val: ChangeEvent<HTMLInputElement>) =>
    onChange && onChange(val.target.value);

  return (
    <label className={combinedClassName} htmlFor={id}>
      <input
        id={id}
        type="radio"
        name={groupName}
        value={value as string}
        checked={value === selectedValue}
        onChange={inputOnChange}
      />
      <span>{children}</span>
    </label>
  );
};

export default RadioButton;
