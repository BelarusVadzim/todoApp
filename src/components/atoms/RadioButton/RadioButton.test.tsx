import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import RadioButton from './RadioButton';

const groupName = 'groupName';
const value = 'value33';

describe('<RadioButton />', () => {
  it('should render properly', () => {
    const component = render(
      <RadioButton className="className" groupName={groupName} value={value}>
        LabelChild
      </RadioButton>
    );
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when onChange raised and onChange handler specified should call expected functions with expected values', () => {
    const onChangeHandler = jest.fn();
    render(
      <RadioButton
        onChange={onChangeHandler}
        groupName={groupName}
        value={value}
      >
        radioButtonText
      </RadioButton>
    );

    const inputElement = screen.getByDisplayValue(value);

    userEvent.type(inputElement, 'typedText');

    expect(onChangeHandler).toBeCalledTimes(1);
    expect(onChangeHandler).toBeCalledWith(value);
  });
});
