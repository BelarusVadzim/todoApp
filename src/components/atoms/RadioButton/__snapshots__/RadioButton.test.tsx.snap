// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should render properly 1`] = `
<DocumentFragment>
  <label
    class="undefined className"
  >
    <input
      name="groupName"
      type="radio"
      value="value33"
    />
    <span>
      LabelChild
    </span>
  </label>
</DocumentFragment>
`;
